File: leetcode/136/SingleNumber.py
# class Solution:
    # def singleNumber(self, nums: List[int]) -> int:
        
        # map = {}
        # for i in range(0,len(nums)):
        #     if nums[i] in map:
        #         map[nums[i]] = map[nums[i]] + 1;
        #     else:
        #         map[nums[i]] = 1;
        
        # for num in nums:
        #     if num in map:
        #         map[num] = map[num] + 1;
        #     else:
        #         map[num] = 1;
            
        # for value in map:
        #     if map[value] == 1:
        #         return value
        # return -1  

class Solution(object):
    def singleNumber(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        a = 0;
        for num in nums:
            a = a ^ num
        return a;

sl = Solution()
print(sl.singleNumber([2,1,2,3,1]))
  

File: leetcode/136/SingleNumber.cpp
#include <iostream>
#include <vector>

using namespace std;

// class Solution {
// public:
//     int singleNumber(vector<int>& nums) {
//         int ret =0;
//         for(int i =0; i < nums.size();i++)
//         {
//             ret = ret ^ nums[i];
//         }
//         return ret;
//     }
// };

// int main()
// {
//     Solution sl;

//     vector<int> nums;
//     nums.push_back(4);   
//     nums.push_back(1);   
//     nums.push_back(2);   
//     nums.push_back(1);   
//     nums.push_back(2);   
//     cout << sl.singleNumber(nums)<<'\n';
//     return 0;
// }
//gcc -o SingleNumber SingleNumber.cpp -lstdc++

class Solution {
public:
    int singleNumber(vector<int>& nums) {
        int ret =0;
        for(int num : nums){
            ret = ret ^num;
        }
        return ret;
    }
};

int main()
{
    Solution sl;

    vector<int> nums = {4,1,2,1,2};    
    cout << sl.singleNumber(nums)<<'\n';
    return 0;
}
//g++ -std=c++17 -o SingleNumber SingleNumber.cpp


